syntax = "proto2";

option java_package = "com.example.tutorial";
option java_outer_classname = "AddressBookProtos";

message Person {
    required string name = 1;
    required int32 id = 2;        // Unique ID number for this person.
    optional string email = 3;

    enum PhoneType {
        MOBILE = 0;
        HOME = 1;
        WORK = 2;
    }

    message PhoneNumber {
        required string number = 1;
        optional PhoneType type = 2 [default = HOME];
    }

    repeated PhoneNumber phone = 4;
}

// Our address book file is just one of these.
message AddressBook {
    repeated Person person = 1;
}

// The addressBook service definition.
service AddressBookService {
    // Sends a greeting
    rpc addPerson (addPersonRequest) returns (addPersonReply) {}
    rpc getAddressBook (getAddressBookRequest) returns (getAddressBookReply){}

}



message getAddressBookRequest {
    optional string addressBookName = 1;
}

message getAddressBookReply {
    required AddressBook addressBook = 1;
}
// The response message containing the greetings
message addPersonReply {
    required string message = 1;
}

message addPersonRequest{
    required Person person = 1;
}